core/src/main/scala/org/broadinstitute/dsde/rawls/entities/EntityService.scala

```diff
@@ -15,31 +17,45 @@ import org.broadinstitute.dsde.rawls.entities.exceptions.{
 }
 import org.broadinstitute.dsde.rawls.metrics.RawlsInstrumented
 import org.broadinstitute.dsde.rawls.model.AttributeUpdateOperations.{AttributeUpdateOperation, EntityUpdateDefinition}
+import org.broadinstitute.dsde.rawls.model.WorkspaceSettingConfig.CompactDataTablesConfig
 import org.broadinstitute.dsde.rawls.model._
 import org.broadinstitute.dsde.rawls.util.TracingUtils.traceFutureWithParent
 import org.broadinstitute.dsde.rawls.util.{AttributeSupport, EntitySupport, JsonFilterUtils, WorkspaceSupport}
-import org.broadinstitute.dsde.rawls.workspace.WorkspaceRepository
+import org.broadinstitute.dsde.rawls.workspace.{WorkspaceRepository, WorkspaceSettingService}
 import org.broadinstitute.dsde.rawls.{RawlsExceptionWithErrorReport, StringValidationUtils}
+import slick.dbio.{DBIO, DBIOAction, Effect, NoStream}
 
 import java.sql.SQLException
+import scala.collection.immutable
 import scala.concurrent.{ExecutionContext, Future}
 
 object EntityService {
   def constructor(dataSource: SlickDataSource,
                   samDAO: SamDAO,
                   workbenchMetricBaseName: String,
                   entityManager: EntityManager,
-                  pageSizeLimit: Int
+                  pageSizeLimit: Int,
+                  workspaceSettingServiceConstructor: Option[RawlsRequestContext => WorkspaceSettingService] =
+                    None // only used for Quicksilver migration
   )(ctx: RawlsRequestContext)(implicit executionContext: ExecutionContext): EntityService =
-    new EntityService(ctx, dataSource, samDAO, entityManager, workbenchMetricBaseName, pageSizeLimit)
+    new EntityService(ctx,
+                      dataSource,
+                      samDAO,
+                      entityManager,
+                      workbenchMetricBaseName,
+                      pageSizeLimit,
+                      workspaceSettingServiceConstructor
+    )
```

Comment:
I don't love adding the `workspaceSettingServiceConstructor` just to support the Quicksilver migration but also didn't want to blow up scope on this PR by creating a brand new service. I'm open to feedback on this choice!